core/src/main/java/io/paradoxical/cassieq/auth/SignedRequestAuthFilter.java

```diff
@@ -0,0 +1,171 @@
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MultivaluedMap;
+import java.io.IOException;
+import java.security.Principal;
+import java.util.EnumSet;
+import java.util.List;
+
+import static com.godaddy.logging.LoggerFactory.getLogger;
+
+@Priority(Priorities.AUTHENTICATION)
+@Builder(builderClassName = "Builder")
+public class SignedRequestAuthFilter<P extends Principal> extends AuthFilter<AuthorizedRequestCredentials, P> {
+    private static final Logger logger = getLogger(SignedRequestAuthFilter.class);
+
+    private final String accountNamePathParameter;
+    private final String keyAuthPrefix;
+
+    public SignedRequestAuthFilter(String accountNamePathParameter, final String keyAuthPrefix) {
+        this.accountNamePathParameter = accountNamePathParameter;
+        this.keyAuthPrefix = keyAuthPrefix;
+    }
+
+    @Override
+    public void filter(final ContainerRequestContext requestContext) throws IOException {
+        try {
+
+            final MultivaluedMap<String, String> pathParameters = requestContext.getUriInfo().getPathParameters();
+            if (pathParameters.containsKey(accountNamePathParameter)) {
+
+                final AccountName accountName = AccountName.valueOf(pathParameters.getFirst(accountNamePathParameter));
+
+                final RequestParameters headerRequestParameters = parseHeaderRequestParameters(requestContext, accountName);
+                final SignedUrlParameters signedUrlParameters = parseSignedUrlParameters(requestContext, accountName);
+
+                final AuthorizedRequestCredentials credentials =
+                        AuthorizedRequestCredentials.builder()
+                                                    .requestParameters(MoreObjects.firstNonNull(headerRequestParameters, signedUrlParameters))
+                                                    .build();
+
+                if (setPrincipal(requestContext, credentials)) {
+                    return;
+                }
+            }
+            else {
+                return;
+            }
+        }
+        catch (IllegalArgumentException e) {
+            logger.warn("Error decoding credentials", e);
+        }
+
+        throw new WebApplicationException(unauthorizedHandler.buildResponse(prefix, realm));
+    }
+
+    private RequestParameters parseHeaderRequestParameters(final ContainerRequestContext requestContext, final AccountName accountName) {
+        final String header = requestContext.getHeaders().getFirst(HttpHeaders.AUTHORIZATION);
+
+        if (!Strings.isNullOrEmpty(header)) {
+            final Splitter splitter = Splitter.on(' ').limit(2).trimResults();
+            final List<String> components = splitter.splitToList(header);
+
+            if (components.size() == 2) {
+                final String authScheme = components.get(0);
+
+                if (keyAuthPrefix.equalsIgnoreCase(authScheme)) {
```

Comment:
So this is when someone calls in with god mode and is using the full key as the auth scheme? can we add some comments as to the difference between to the two if statements?